BGL/include/CGAL/boost/graph/Connected_component_graph.h

```diff
@@ -0,0 +1,602 @@
+    typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor h = halfedge(v, w);
+    typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::halfedge_descriptor hcirc = h;
+    do
+    {
+        if(is_valid(hcirc, w))
+            ++v_deg;
+        hcirc = opposite(next(hcirc, w.graph()), w.graph());
+    }while(hcirc != h);
+    return v_deg;
+}
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits<Graph>::degree_size_type
+out_degree(typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor v,
+           const Connected_component_graph<Graph, FaceComponentMap>& w)
+{
+    CGAL_assertion(is_valid(v, w));
+    return std::distance(out_edges(v, w).first ,out_edges(v, w).second);
+}
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits<Graph>::degree_size_type
+in_degree(typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor v,
+          const Connected_component_graph<Graph, FaceComponentMap>& w)
+{
+    CGAL_assertion(is_valid(v, w));
+    return std::distance(in_edges(v, w).first ,in_edges(v, w).second);
+}
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor
+source(typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::edge_descriptor e,
+       const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    CGAL_assertion(is_valid(e, w));
+    return source(e, w.graph());
+}
+
+template <class Graph, typename FaceComponentMap >
+typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor
+target(typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::edge_descriptor e,
+       const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    CGAL_assertion(is_valid(e, w));
+    return target(e, w.graph());
+}
+
+template <class Graph, typename FaceComponentMap >
+std::pair<typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::edge_descriptor, bool>
+edge(typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor u,
+     typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_descriptor v,
+     const Connected_component_graph<Graph, FaceComponentMap> & w)
+{
+    CGAL_assertion(is_valid(u, w) && is_valid(v, w));
+    typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::edge_descriptor e = edge(u, v, w.graph()).first;
+    bool res = is_valid(e, w);
+    return std::make_pair(e, res);
+}
+
+
+template <class Graph, typename FaceComponentMap >
+std::pair<typename boost::graph_traits<Connected_component_graph<Graph, FaceComponentMap> >::vertex_iterator,
```

Comment:
Replace the `std::par` with a `CGAL::Iterator_range`. 
Please do this also in the `Gwdwg`.